packages/orchestration/README.md

```diff
@@ -17,16 +17,145 @@ $ npm install @sap-ai-sdk/orchestration
+
+### Prerequisites
+
+Ensure the following are in place:
+
+- At least one orchestration-compatible deployment for a generative AI model is running.
+  - You can use the [`DeploymentApi`](../ai-api/README.md#deploymentapi) from `@sap-ai-sdk/ai-api` to deploy a model to the SAP generative AI hub. For more information, see [here](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/create-deployment-for-generative-ai-model-in-sap-ai-core).
+- A deployment for orchestration is created. For details, see [here](https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/create-deployment-for-orchestration).
+- The `@sap-ai-sdk/orchestration` package is installed in your project.
+
+### Orchestration Client
+
+Use the `OrchestrationClient` to configure different modules like templating and content filtering along with sending chat completion requests to an orchestration compatible generative AI model.
+
+You can pass configuration information like for example the LLM to be used for inference and template, as a parameter to the client.
+
+#### Orchestration client with Templating
+
+Use the Orchestration client with templating to pass a prompt with placeholders, that can be filled during a chat completion request with the input parameters.
+
+This feature enables you to vary the prompt based on the input parameters.
+
+```TS
+import { OrchestrationClient } from '@sap-ai-sdk/orchestration';
+
+const orchestrationClient = new OrchestrationClient({
+  llm: {
+    model_name: 'gpt-4-32k',
+    model_params: { max_tokens: 50, temperature: 0.1 }
+  },
+  templating: {
+    template: [
+      { role: 'user', content: 'What is the capital of {{?country}}?' }
+    ]
+  }
+});
+
+const response = await orchestrationClient.chatCompletion({
+  inputParams: { country: 'France' }
+});
+
+const responseContent = response.getContent();
+```
+
+It is possible to provide a history of a conversation to the model.
+Use the following snippet to send a chat completion request with history and a system message:
+
+```TS
+import { OrchestrationClient } from '@sap-ai-sdk/orchestration';
+
+const orchestrationClient = new OrchestrationClient({
+  llm: {
+    model_name: 'gpt-4-32k',
+    model_params: { max_tokens: 50, temperature: 0.1 }
+  },
+  templating: {
+    template: [
+      { role: 'user', content: 'What is my name?' }
+    ],
+    messages_history: [
+      {
+        role: 'system',
+        content: 'You are a helpful assistant who remembers all details the user shares with you.'
+      },
+      {
+        role: 'user',
+        content: 'Hi! I am Bob'
+      },
+      {
+        role: 'assistant',
+        content: 'Hi Bob, nice to meet you! I am an AI assistant. I will remember that your name is Bob as we continue our conversation.'
+      }
+    ]
+  }
+});
+
+const responseContent = await orchestrationClient.chatCompletion().getContent();
```

Comment:
[req] This is still under templating, however not related to templating. Consider adding a separate title.

Reply:
I would prefer to split it like this:
```
const response = await orchestrationClient.chatCompletion();
const responseContent = response.getContent();
```

And leave it this way, as I feel code is self explanatory and explaining it will make it super verbose.

Reply:
Ah, I remember that the orchestration response content is a bit complicated. 
I will have to check it out in a E2E test locally, before explaining it.